### PR TITLE
feat: CEL Validation in BackendTrafficPolicy

### LINKED ISSUE #2066: Add more CEL Validation in BackendTrafficPolicy
*Description*:
>Describe the issue.

* Disable `targetRef.sectionName` for now
* Make sure `consistentHash` is non nil when type of load balancer is set to `ConsistentHash`

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by slayer321**
**What type of PR is this?**

feat: CEL Validation in BackendTrafficPolicy
**What this PR does / why we need it**:
CEL Validation in BackendTrafficPolicy for load balancer field

**Which issue(s) this PR fixes**:

Fixes #2066

PS: issues also mentioned that we need to disable `targetRef.sectionName` field . But I found that it was already Implemented by Alice.

### CODE REVIEW

**Review by arkodg — APPROVED** [review]
LGTM thanks !
cc @zirain @AliceProxy

File: test/cel-validation/backendtrafficpolicy_test.go [hunk 1]
```diff
@@ -151,6 +151,47 @@ func TestBackendTrafficPolicyTarget(t *testing.T) {
 				"spec.targetRef: Invalid value: \"object\": this policy does not yet support the sectionName field",
 			},
 		},
+		{
```
Comment (zirain) [inline test/cel-validation/backendtrafficpolicy_test.go:154]: can you add a test which LoadBalancer type is `LeastRequest` and leave ConsistentHash nil.

File: api/v1alpha1/loadbalancer_types.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ package v1alpha1
 
 // LoadBalancer defines the load balancer policy to be applied.
 // +union
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConsistentHash' && has(self.consistentHash)",message="If LoadBalancer type is consistentHash, consistentHash field needs to be set."
```
Comment (zirain) [inline api/v1alpha1/loadbalancer_types.go:11]:
```suggestion
// +kubebuilder:validation:XValidation:rule="self.type == 'ConsistentHash' ? has(self.consistentHash) : !has(self.consistentHash)",message="If LoadBalancer type is consistentHash, consistentHash field needs to be set."
```

**Review by zirain — APPROVED** [review]
defer to @AliceProxy for another look

**Review by arkodg — APPROVED** [review]
Alice is away, merging for now, thanks @slayer321 !
